Code/Framework/AzCore/AzCore/UnitTest/UnitTest.h

```diff
@@ -181,9 +182,37 @@ namespace UnitTest
         }
     };
 
+    class TraceBusHook;
+
+    // Helper class to automatically re-enable the TraceBusHook when going out of scope
+    // This is necessary because the TraceBusHook is part of the Environment which does not reset after each test run
+    // So we need to ensure the bus is always reconnected between tests
+    struct TraceBusHookUniqueDisableToken
+    {
+        TraceBusHookUniqueDisableToken(TraceBusHook* traceBusHook);
+        ~TraceBusHookUniqueDisableToken();
+
+        void ReEnable();
```

Comment:
This ReEnable function shouldn't be needed.
Using the paradigm of Resource Acquisition is Initialization and having the `TraceBusHookEnvironmentInterface` return a `TraceBusHookUniqueDisableToken`, the all the user would have to do is invoke the `TraceBusHookEnvironmentInterfaceDisable` function and then clear it when it is no longer needed

Reply:
I guess that works.  I kinda liked how ReEnable read but this works too

Reply:
Yeah, +1 on using a simple RAII style interface for this.. the ReEnable call is superfluous and adds unnecessary complexity to the usage of this class.